libgsf.yaml

```diff
@@ -43,6 +43,12 @@ pipeline:
       expected-commit: ea9d8cd1369661bf62d0476474700a9e0887c812
       tag: LIBGSF_${{vars.mangled-package-version}}
 
+  - runs: |
+      git fetch origin master:master
+      # Fix failure to build from source against new libxml2, without
+      # nanohttp support. Ignore merge conflict in NEWS.
+      git cherry-pick -x 5d4bb55095d3d6ef793c1908a88504183e28644c || true
+
```

Comment:
the `|| true` could fail in any way.  Unlikely but worst case is leaving buggy code that compiles.

I think I'd prefer a hand edited patch to remove the NEWS.  

Another option is https://github.com/wolfi-dev/os/pull/25657 paired with something like:
```
    git format-patch --stdout -1 5d4bb55095d3d6ef793c1908a88504183e28644c > cp.diff
    filterdiff -p1 --exclude=NEWS < cp.diff > filtered.diff
    git apply filtered.diff
    rm -f cp.diff filtered.diff
```

I don't love it. there is 
https://stackoverflow.com/questions/5717026/how-to-cherry-pick-only-changes-to-certain-files
which mentions `git show SHA -- file1 file2`, but that is include rather than exclude. :cry:


another option that I think I like the best

```
# Fix failure to build from source against new libxml2, without
# nanohttp support. Ignore merge conflict in NEWS.
git cherry-pick -x 5d4bb55095d3d6ef793c1908a88504183e28644c || true
git checkout HEAD NEWS
git cherry-pick --continue --no-edit
```

The improvement here is that the cherry-pick `--continue` will fail if there were other changes that were not cleanly applied.

If the NEWs issue is a repeating pattern, we could add support in some way for that into checkout's cherry-picks.